core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractSequencerReference.java

```diff
@@ -23,34 +23,64 @@
 
 package uk.ac.bbsrc.tgac.miso.core.data;
 
-import java.io.IOException;
 import java.net.InetAddress;
+import java.net.UnknownHostException;
 import java.util.Date;
 
+import javax.persistence.Column;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.MappedSuperclass;
+import javax.persistence.OneToOne;
+import javax.persistence.Transient;
+
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import uk.ac.bbsrc.tgac.miso.core.data.impl.PlatformImpl;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.SequencerReferenceImpl;
+
 /**
  * Abstract class to provide basic methods to encapsulate a reference to a physical machine attached to a sequencer
  * 
  * @author Rob Davey
  * @since 0.0.2
  */
+@MappedSuperclass
 public abstract class AbstractSequencerReference implements SequencerReference {
   protected static final Logger log = LoggerFactory.getLogger(AbstractSequencerReference.class);
 
   public static final Long UNSAVED_ID = 0L;
 
+  @Id
+  @GeneratedValue(strategy = GenerationType.AUTO)
+  @Column(name = "referenceId")
   private long id = AbstractSequencerReference.UNSAVED_ID;
+
+  @Column(nullable = false)
   private String name;
+
+  @ManyToOne(targetEntity = PlatformImpl.class)
+  @JoinColumn(name = "platformId", nullable = false)
   private Platform platform;
-  private Boolean available;
-  private InetAddress ip;
+
+  private String ip;
+
   private String serialNumber;
   private Date dateCommissioned;
   private Date dateDecommissioned = null;
+
+  @OneToOne(targetEntity = SequencerReferenceImpl.class)
```

Comment:
The current model/dao will cause an exception if this isn't 1:1, but it could be ManyToOne, and it wouldn't exactly be wrong. If the relationship is made unidirectional (by removing preUpgradSequencerReference), this should be ManyToOne; otherwise, OneToOne is probably more correct